lib/msf/core/payload.rb

```diff
@@ -295,7 +295,7 @@ def handler_klass
   # Returns the session class that is associated with this payload and will
   # be used to create a session as necessary.
   #
-  def session
+  def session_klass
```

Comment:
Aye it's a big concern. It begs the questions:

* Why was it like this before?
* What is the potential damage?
* Is using the code in this way, resulting in me needing to change this stuff, the right approach at all?